system/jlib/jstats.cpp

```diff
@@ -4011,6 +4051,277 @@ void verifyStatisticFunctions()
     }
 }
 
+class StatsAggregator : public CInterface
+{
+    class ScopeId: public StatsScopeId
+    {
+    public:
+        ScopeId(StatisticScopeType _scopeType=SSTglobal, unsigned _id=0) : StatsScopeId(_scopeType, _id) {}
+        ScopeId(const ScopeId & _id) : StatsScopeId(_id.scopeType, _id.id, _id.extra) {}
+        ScopeId(const StatsScopeId & other) : StatsScopeId(other) {}
+        size_t operator()(const ScopeId & _scopeId) const
+        {
+            return std::hash<unsigned>{}((unsigned)(_scopeId.id|(_scopeId.scopeType<<16)));
+        }
+        bool operator()(const ScopeId & lhs, const ScopeId & rhs) const
+        {
+            return (lhs.scopeType==rhs.scopeType) && (lhs.id==rhs.id);
+        }
+    };
+    struct StatisticKindFunc  // funcs required by unordered_map<StatisticKind,..>
+    {
+        std::size_t operator()(const StatisticKind & sk) const
+        {
+            return std::hash<unsigned>{}((unsigned)sk);
+        }
+        bool operator()(const StatisticKind & lhs, const StatisticKind & rhs) const
```

Comment:
Is this actually needed, I would expect the default implementation to work?  If it is needed it would be clearer as a separate StatisticKindComparer class.